samples/OpenIdConnectSample/Startup.cs

```diff
@@ -151,15 +162,76 @@ await context.SignOutAsync(OpenIdConnectDefaults.AuthenticationScheme, new Authe
                     return;
                 }
 
-                await WriteHtmlAsync(context.Response, async response =>
+                if (context.Request.Path.Equals("/refresh"))
+                {
+                    var refreshToken = props.GetTokenValue("refresh_token");
+
+                    if (string.IsNullOrEmpty(refreshToken))
+                    {
+                        await WriteHtmlAsync(response, async res =>
+                        {
+                            await res.WriteAsync($"No refresh_token is available.<br>");
+                            await res.WriteAsync("<a class=\"btn btn-link\" href=\"/signout\">Sign Out</a>");
+                        });
+
+                        return;
+                    }
+
+                    var handler = (AuthenticationHandler<OpenIdConnectOptions>)await authSchemes.GetHandlerAsync(context, OpenIdConnectDefaults.AuthenticationScheme);
```

Comment:
If you aren't using the handler yet, I wouldn't resolve the handler, I'd just get the options from IOptionsManager instead